consumer/src/main/java/com/autostreams/pulsar/ConsumerPropertyLoader.java

```diff
@@ -0,0 +1,154 @@
+package com.autostreams.pulsar;
+
+import com.autostreams.utils.fileutils.FileUtils;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * Represents an object responsible for handling and loading configurations for the consumer.
+ *
+ * @version 1.0
+ * @since 1.0
+ */
+public class ConsumerPropertyLoader {
+    private static final String CONFIG_NAME = "consumerconfig.properties";
+    private final ArrayList<ConfigurationNamePair> configVariableKeys = new ArrayList<>();
+    private final Set<String> topics = new HashSet<>();
+    private static final String HOST_PROPERTY_VARIABLE_NAME = "url";
+    private static final String HOST_ENVIRONMENT_VARIABLE_NAME = "URL";
+
+    private record ConfigurationNamePair(String propertyVariableName,
+                                         String environmentVariableName) {
+    }
+
+    /**
+     * Gets the available consumer configuration. Environment variables are prioritized over
+     * the .properties file, meaning that if both exist the environment variables will be used.
+     *
+     * @return String-Object map containing configuration keys as strings and their values as
+     *         objects
+     */
+    public Map<String, Object> getConsumerConfiguration() {
+        this.setConfigVariableKeys();
+        String firstVariableKey = this.configVariableKeys.get(0).environmentVariableName;
```

Comment:
Does this only check the very first environment variable? What about all the rest if they are not present?